pkg/build/api/v1/types.go

```diff
@@ -176,8 +176,89 @@ type BuildStatus struct {
 
 	// output describes the Docker image the build has produced.
 	Output BuildStatusOutput `json:"output,omitempty" protobuf:"bytes,10,opt,name=output"`
+
+	// steps contains details about each build step.
+	Stages Stages `json:"stages,omitempty" protobuf:"bytes,11,opt,name=stages"`
+}
+
+// Stages is a slice of build stages that have been recorded.
+type Stages []StageInfo
+
+// StageInfo contains details about a build stage.
+type StageInfo struct {
+	// StageName is the identifier for each build stage.
+	StageName StageName `json:"stageName,omitempty" protobuf:"bytes,1,opt,name=stageName"`
```

Comment:
we have constants defined like StageFetchInputs StageName = "FetchInputs", 
so "Name" would be pretty generic there?